Tweak docs to make Dockerized build more visible (#3875)
DataDog/datadog-agent: docs/dev/agent_omnibus.md

# Build the Agent packages

Agent packages for all the supported platforms are built using
[Omnibus](https://github.com/chef/omnibus), which can be run via `invoke` tasks.

> Omnibus will create a package for your operating system, so you'll get a DEB

Reviewer: ```suggestion
Omnibus creates a package for your operating system, so you'll get a DEB
```

package on Debian-based distros, an RPM package on distros that use RPM, an MSI
installer on Windows, or a `.pkg` package bundled in a DMG archive on Mac.

For Linux, we provide Docker images (one to build DEB packages and one for RPM),
with the build dependencies installed, so you don't have to install them on your system.

## Building inside Docker (Linux only, recommended)

> Let's see in detail how to use the provided Docker images to build a DEB or RPM

Reviewer: ```suggestion
Use the provided Docker images to build a DEB or RPM
```

package for Linux. You need to have Docker already running on your machine.

From the `datadog-agent` source folder, use the following command to run the
`agent.omnibus-build` task in a Docker container:

```
docker run -v "$PWD:/datadog-agent" -v "/tmp/omnibus:/omnibus" -v "/tmp/opt/datadog-agent:/opt/datadog-agent" -v"/tmp/gems:/gems" --workdir=/datadog-agent datadog/agent-buildimages-deb_x64 inv -e agent.omnibus-build --base-dir=/omnibus --gem-path=/gems
```

The container will share 3 volumes with the host to avoid starting from scratch
at each Omnibus run:

 * `/tmp/omnibus`, containing the Omnibus base dir
 * `/tmp/opt/datadog-agent`, containing the Omnibus installation dir
 * `/tmp/gems`, containing all the ruby gems installed with Bundler

> Note that youu can change `deb_x64` for `rpm_x64` to get an RPM package instead.

Reviewer 2: ```suggestion
Note that you can change `deb_x64` for `rpm_x64` to get an RPM package instead.
```

If you want to find the Dockerfiles for these images, they are available in the
[datadog-agent-buildimages](https://github.com/DataDog/datadog-agent-buildimages) git repo.
To build them from scratch, you can do so like this:

```
docker build -t datadog-agent-buildimages:deb_x64 -f deb-x64/Dockerfile .
```

If the build images crash when you run them on modern Linux distributions, you might be
affected by [this bug](https://github.com/moby/moby/issues/28705).

> ## Building on your system (Linux and Mac)

Reviewer 2: Add here the requirements part of the `Prepare the dev environment` section that was removed, since you do need these to run Omnibus locally?

Author: You also need python and pip, clang, systemd headers, and a ton of other dependencies... I found the list so incomplete I thought it was better to remove it.
